File: api/src/main/java/net/tachyon/coordinate/BlockFace.java
package net.tachyon.coordinate;

import org.jetbrains.annotations.NotNull;

public enum BlockFace {
    BOTTOM(Direction.DOWN),
    TOP(Direction.UP),
    NORTH(Direction.NORTH),
    SOUTH(Direction.SOUTH),
    WEST(Direction.WEST),
    EAST(Direction.EAST);

    private final Direction direction;

    BlockFace(Direction direction) {
        this.direction = direction;
    }

    public Direction toDirection() {
        return direction;
    }

    @NotNull
    public BlockFace getOppositeFace() {
        return switch (this) {
            case BOTTOM -> TOP;
            case TOP -> BOTTOM;
            case NORTH -> SOUTH;
            case SOUTH -> NORTH;
            case WEST -> EAST;
            case EAST -> WEST;
        };
    }
}

File: Server/World/Block/BlockFace.cs
using Server.Position;

namespace Server.World.Block;

public class BlockFace
{

    public static readonly BlockFace BOTTOM = new(Direction.DOWN);
    public static readonly BlockFace TOP = new(Direction.UP);
    public static readonly BlockFace NORTH = new(Direction.NORTH);
    public static readonly BlockFace SOUTH = new(Direction.SOUTH);
    public static readonly BlockFace WEST = new(Direction.WEST);
    public static readonly BlockFace EAST = new(Direction.EAST);

    public readonly Direction Direction;
    
    private BlockFace(Direction direction)
    {
        Direction = direction;
    }

}